app/frontend/wallet/shelley/shelley-transaction-planner.ts

```diff
@@ -166,11 +165,14 @@ export const computeMinUTxOLovelaceAmount = (tokenBundle: TokenBundle): Lovelace
   // NOTE: should be 29 but a bug in Haskell set this to 27
   const adaOnlyUtxoSize = utxoEntrySizeWithoutVal + coinSize
 
-  const distinctAssets = distinct(tokenBundle.map(({assetName}) => assetName))
+  // this ensures there are only distinct (policyId, assetName) pairs
```

Comment:
I would cover this with unit tests - it's a function without side-effects so an easy win in terms of testing, yet it has quite a non-trivial logic which is easy to break/introduce regressions as we modify it/implement further possible fixes